src/transformStateWithClones.js

```diff
@@ -8,6 +8,40 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const arrayWithObjects = [];
+  let cloneState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        cloneState = {
+          ...cloneState,
+          ...action.extraData,
+        };
+
+        break;
+
+      case 'removeProperties':
+        cloneState = removeProperties(cloneState, action.keysToRemove);
+
+        break;
+      case 'clear':
+        cloneState = {};
+
+        break;
+    }
```

Comment:
[CODE STYLE]: Use a default case in your switch statement to handle any unexpected action types. This helps in error handling and ensures that the function behaves predictably even with incorrect input.